File: examples/typescript/publisher.ts
async function main() {
  const gatewayUrl = process.env.GATEWAY_URL ?? 'http://127.0.0.1:8787'
  const topic = process.env.TOPIC ?? 'alerts'
  const eventId = process.env.EVENT_ID ?? `ts-event-${Date.now()}`
  const apiKey = process.env.API_KEY ?? 'dev-key'
  const payload = process.env.PAYLOAD ?? 'hello from typescript publisher'

  const response = await fetch(`${gatewayUrl}/v1/publish`, {
    method: 'POST',
    headers: {
      Authorization: `Bearer ${apiKey}`,
      'x-sse-topic': topic,
      'x-sse-id': eventId,
    },
    body: payload,
  })

  if (response.status !== 202) {
    throw new Error(`publish failed: ${response.status} ${await response.text()}`)
  }

  console.log(await response.text())
}

main().catch((error: unknown) => {
  console.error(error)
  process.exit(1)
})

File: examples/rust/src/bin/publisher.rs
use std::env;

fn getenv(key: &str, fallback: &str) -> String {
    env::var(key).unwrap_or_else(|_| fallback.to_string())
}

fn main() {
    let gateway_url = getenv("GATEWAY_URL", "http://127.0.0.1:8787");
    let topic = getenv("TOPIC", "alerts");
    let event_id = getenv("EVENT_ID", "rust-event-1");
    let api_key = getenv("API_KEY", "dev-key");
    let payload = getenv("PAYLOAD", "hello from rust publisher");

    let client = reqwest::blocking::Client::new();
    let response = client
        .post(format!("{}/v1/publish", gateway_url))
        .header("Authorization", format!("Bearer {}", api_key))
        .header("x-sse-topic", topic)
        .header("x-sse-id", event_id)
        .body(payload)
        .send()
        .expect("publish request failed");

    let status = response.status();
    let body = response.text().expect("failed to read publish response");

    if status.as_u16() != 202 {
        panic!("publish failed: {} {}", status.as_u16(), body);
    }

    println!("{}", body);
}
